From cb3de39eeff19ea09762617c267a7676f55a3372 Mon Sep 17 00:00:00 2001
From: Radhey Shyam Pandey <radhey.shyam.pandey@xilinx.com>
Date: Tue, 19 May 2015 17:16:53 +0530
Subject: [PATCH 528/793] gpu: drm: xylon: Return error if i2c client driver
 is NULL

Return error if i2c_client driver instance is NULL.
Incase i2c encoder(adv7511) probe fails not checking
it will result in NULL pointer dereference.

<snip>
Unable to handle kernel NULL pointer dereference at virtual address 00000050
pgd = 40004000
[00000050] *pgd=00000000
Internal error: Oops - BUG: 17 [#1] PREEMPT SMP ARM
Modules linked in:
CPU: 0 PID: 690 Comm: kworker/u4:2 Not tainted 3.19.0-xilinx-13711-g2b55e97-dirty #49
Hardware name: Xilinx Zynq Platform
Workqueue: deferwq deferred_probe_work_func
task: 771ab100 ti: 72a42000 task.ti: 72a42000
PC is at xylon_drm_encoder_create+0xf4/0x188
LR is at xylon_drm_encoder_create+0xf4/0x188

Signed-off-by: Radhey Shyam Pandey <radheys@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Tested-by: Christian Kohn <christian.kohn@xilinx.com>
---
 drivers/gpu/drm/xylon/xylon_encoder.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/gpu/drm/xylon/xylon_encoder.c b/drivers/gpu/drm/xylon/xylon_encoder.c
index 7bad2b9..b2156ab 100644
--- a/drivers/gpu/drm/xylon/xylon_encoder.c
+++ b/drivers/gpu/drm/xylon/xylon_encoder.c
@@ -187,7 +187,7 @@ struct drm_encoder *xylon_drm_encoder_create(struct drm_device *dev)
 
 	encoder->client = of_find_i2c_device_by_node(sub_node);
 	of_node_put(sub_node);
-	if (!encoder->client) {
+	if (!encoder->client || !encoder->client->dev.driver) {
 		DRM_INFO("failed find encoder\n");
 		return ERR_PTR(-EPROBE_DEFER);
 	}
-- 
1.7.5.4

